### PR TITLE
[feat] Add function to retrieve paginated list of child users

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Add beta service for User
- Add `AllChildren` and `GetNextPageOfChildren` functions to beta User service
- Drop `BaseUser` class in favor of explicit re-implementation in `User` and `ReferralCustomer` classes (flatten out inheritance tree)

# Testing

- Add unit tests, cassettes
- Fix straggling back CRUD decorators

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: EasyPost.Tests/ServicesTests/WithParameters/Beta/UserServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using EasyPost.Tests._Utilities;
+using EasyPost.Tests._Utilities.Attributes;
+using EasyPost.Utilities.Internal.Attributes;
+using Xunit;
+
+namespace EasyPost.Tests.ServicesTests.WithParameters.Beta
+{
+    public class UserServiceTests : UnitTest
+    {
+        public UserServiceTests() : base("beta_user_service_with_parameters", TestUtils.ApiKey.Production) =>
```
Comment (Justintime50) [inline EasyPost.Tests/ServicesTests/WithParameters/Beta/UserServiceTest.cs:14]: This isn't beta anymore right? It's also in the `Beta` directory?

```suggestion
        public UserServiceTests() : base("user_service_with_parameters", TestUtils.ApiKey.Production) =>
```

Comment (nwithan8) [reply EasyPost.Tests/ServicesTests/WithParameters/Beta/UserServiceTest.cs:14]: Good catch, forgot to delete the file once I migrated stuff out of it.

File: EasyPost/Models/API/ChildUserCollection.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+/// </summary>
+public class ChildUserCollection : PaginatedCollection<User>
+{
+    #region JSON Properties
+
+    /// <summary>
+    ///     The Child <see cref="User"/>s in the collection.
+    /// </summary>
+    [JsonProperty("children")]
+    public List<User>? Children { get; set; }
```
Comment (jchen293) [inline EasyPost/Models/API/ChildUserCollection.cs:20]: User and Children have different attributes, the list of users here doesn't seem correct. We should probably need to create a child model in C# lib.

Comment (nwithan8) [reply EasyPost/Models/API/ChildUserCollection.cs:20]: After discussion, we decided not to make a separate `ChildUser` class due to serialization issues.

However, I took the opportunity to clean up some unnecessary inheritance by dropping the `BaseUser` class.

**Review by jchen293 — APPROVED** [review]

**Pull request merged**